cms/release_calendar/forms.py

```diff
@@ -119,3 +120,71 @@ def validate_bundle_not_pending_publication(self, status: str) -> None:
             bundle_str,
         )
         raise ValidationError({"status": message})
+
+    def clean_changes_to_release_date(self) -> StreamValue:
+        changes_to_release_date: StreamValue = self.cleaned_data.get("changes_to_release_date")
+
+        if self.instance.pk is None:
+            if changes_to_release_date:
+                raise ValidationError(
+                    "You cannot create a change to release date for a page that has not been published yet."
+                )
+            return changes_to_release_date
+
+        published_page = PageLogEntry.objects.filter(page=self.instance, action="wagtail.publish").exists()
+        if changes_to_release_date and not published_page:
+            self.add_error(
+                "changes_to_release_date",
+                ValidationError(
+                    "You cannot create a change to release date for a page that has not been published yet."
+                ),
+            )
+        old_frozen_versions = [
+            date_change_log.value["version_id"]
+            for date_change_log in self.instance.changes_to_release_date
+            if date_change_log.value["frozen"]
+        ]
+        new_frozen_versions = [
+            date_change_log.value["version_id"]
+            for date_change_log in changes_to_release_date
+            if date_change_log.value["frozen"]
+        ]
+
+        if old_frozen_versions != new_frozen_versions:
+            raise ValidationError("You cannot remove a change to release date that has already been published.")
+
+        latest_frozen_date = max(
+            (
+                date_change_log.value["previous_date"]
+                for date_change_log in changes_to_release_date
+                if date_change_log.value["frozen"]
+            ),
+            default=None,
+        )
+
+        new_date_change_log = None
+        latest_date_change_log_version = 0
+
+        for date_change_log in changes_to_release_date:
+            if not date_change_log.value["frozen"]:
+                if new_date_change_log:
+                    raise ValidationError("Only one new change to release date can be published at a time")
```

Comment:
I was able to add multiple notes by merely saving the draft, not even changing the release date field. So I think we need a bit more validation